src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1108,10 +1108,11 @@ int TorrentHandle::timeSinceDownload() const
 
 int TorrentHandle::timeSinceActivity() const
 {
-    if (m_nativeStatus.time_since_upload < m_nativeStatus.time_since_download)
-        return m_nativeStatus.time_since_upload;
-    else
+    if ((m_nativeStatus.time_since_download >= 0) && (m_nativeStatus.time_since_upload < 0))
```

Comment:
I would write as follows:
```c++
return (std::signbit(m_nativeStatus.time_since_upload) != std::signbit(m_nativeStatus.time_since_download)) ?
   std::max(m_nativeStatus.time_since_upload, m_nativeStatus.time_since_download) :
   std::min(m_nativeStatus.time_since_upload, m_nativeStatus.time_since_download);
```
to highlight the condition.

Reply:
done.